state-chain/traits/src/lib.rs

```diff
@@ -833,6 +833,25 @@ pub trait AccountRoleRegistry<T: frame_system::Config> {
 	}
 }
 
+pub trait DeregistrationCheck {
+	type AccountId;
+	type Error: Into<DispatchError>;
+	fn check(account_id: &Self::AccountId) -> Result<(), Self::Error>;
+}
+
+impl<A: DeregistrationCheck, B: DeregistrationCheck<AccountId = A::AccountId>> DeregistrationCheck
```

Comment:
This isn't used anywhere? Was the intention to add another Check in this PR?

Reply:
I don't think so - it was more that the trait doesn't make much sense unless you can compose multiple. I don't mind removing it.

Reply:
Ok just checking, I don't mind leaving it anyway. up to you